Assignment 5.Rmd

```diff
@@ -60,18 +65,26 @@ plot(score_ctree)
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
 
+#The types of advice given by teachers depends on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required. For students who require no hints, there are 145 students who answered less than 85 questions and 76 who answered more than 85 questions. For students who require hints, there are 46 who require more than 12 hints; 45 who require less than 12 hints and score higher than 0.629 and 66 who require less than 12 hints but score lower than 0.629. Teacher should pay most closely attention to students who require more than 12 hints, and less than 12 hints but score lower than 0.629. 
+
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
-
-#Generate predicted advice for new students based on tree generated from old students
+D2 <- read.csv("intelligent_tutor_new.csv", header = TRUE)
 
+#Generate predicted advice for new students based on tree generated from old studentsq
 D2$prediction <- predict(score_ctree, D2)
-
 ``` 
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
+
+```{r}
+D2$actual <- ifelse(D2$prediction == "no action", 1, 0)
+sum(D2$actual)
+#difference: 200 - 116 = 84
+#accuracy: 116 / 200 = 0.58
```

Comment:
Great work, can you provide some explanation for your code though?

Reply:
Hi Charles, I pushed another work below this, with some explanation included. Please review it, thanks!